src/NServiceBus.Core.Analyzer/AwaitOrCaptureTasksAnalyzer.cs

```diff
@@ -0,0 +1,100 @@
+
+            "NServiceBus.IPipelineContextExtensions.Send",
+            "NServiceBus.IPipelineContextExtensions.Send`1",
+            "NServiceBus.IPipelineContextExtensions.SendLocal",
+            "NServiceBus.IPipelineContextExtensions.SendLocal`1",
+            "NServiceBus.IPipelineContextExtensions.Publish",
+            "NServiceBus.IPipelineContextExtensions.Publish`1",
+
+            "NServiceBus.IMessageSession.Send",
+            "NServiceBus.IMessageSession.Send`1",
+            "NServiceBus.IMessageSession.Publish",
+            "NServiceBus.IMessageSession.Publish`1",
+            "NServiceBus.IMessageSession.Subscribe",
+            "NServiceBus.IMessageSession.Unsubscribe",
+
+            "NServiceBus.IMessageSessionExtensions.Send",
+            "NServiceBus.IMessageSessionExtensions.Send`1",
+            "NServiceBus.IMessageSessionExtensions.SendLocal",
+            "NServiceBus.IMessageSessionExtensions.SendLocal`1",
+            "NServiceBus.IMessageSessionExtensions.Publish",
+            "NServiceBus.IMessageSessionExtensions.Publish`1",
+            "NServiceBus.IMessageSessionExtensions.Subscribe",
+            "NServiceBus.IMessageSessionExtensions.Subscribe`1",
+            "NServiceBus.IMessageSessionExtensions.Unsubscribe",
+            "NServiceBus.IMessageSessionExtensions.Unsubscribe`1",
+            
+            "NServiceBus.Saga.RequestTimeout`1",
+            "NServiceBus.Saga.ReplyToOriginator",
+            
+            "NServiceBus.Endpoint.Create",
+            "NServiceBus.IStartableEndpoint.Start",
+            "NServiceBus.IEndpointInstance.Stop");
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(diagnostic);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(Analyze, SyntaxKind.InvocationExpression);
+
+        void Analyze(SyntaxNodeAnalysisContext context)
+        {
+            var call = context.Node as InvocationExpressionSyntax;
+            if (call == null)
+            {
+                return;
+            }
+
+            if (!(call.Parent is ExpressionStatementSyntax))
+            {
+                return;
+            }
+
+            if (IsInAsyncMethod(call))
+            {
+                return;
+            }
+
+            var methodSymbol = context.SemanticModel.GetSymbolInfo(call).Symbol as IMethodSymbol;
+            if (methodSymbol != null && methods.Contains(methodSymbol.GetFullNameWithArity()))
+            {
+                context.ReportDiagnostic(Diagnostic.Create(diagnostic, call.GetLocation(), call.ToString()));
+            }
+        }
+
+        static bool IsInAsyncMethod(SyntaxNode node)
+            => HasAsyncContext(node.Parent);
+
+        static bool HasAsyncContext(SyntaxNode node) =>
+            node != null && (IsAsync(node as MethodDeclarationSyntax) || HasAsyncContext(node.Parent));
```

Comment:
It looks like IsAsync would get called with null value (which means it is known the output will be false) in all except the "I have made it to the method declaration" case. Would it be better to do a `node is MethodDeclarationSyntax` before calling out to the method? Or possible to suggest inlining to the compiler?

Reply:
@DavidBoike agreed, this should be covered by this PR https://github.com/Particular/NServiceBus/pull/5163 ?